### PR TITLE
feat: Adds OpenAPI spec

### PR DESCRIPTION
**Pull request by soluchok**
Adds OpenAPI spec.

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -68,12 +73,60 @@ depend:
...
+	scripts/generate-openapi-spec.sh
+
+.PHONY: generate-openapi-demo-specs
+generate-openapi-demo-specs: clean generate-openapi-spec agent-rest-docker
+	@echo "Generate demo agent rest controller API specifications using Open API"
+	@SPEC_PATH=${OPENAPI_SPEC_PATH} OPENAPI_DEMO_PATH=deployments/demo/openapi \
+    	DOCKER_IMAGE=$(OPENAPI_DOCKER_IMG) DOCKER_IMAGE_VERSION=$(OPENAPI_DOCKER_IMG_VERSION)  \
+    	scripts/generate-openapi-demo-specs.sh
+
+generate-test-keys: clean
```
Comment (llorllale) [inline Makefile:91]: do we really need this to generate an open-api spec?

File: cmd/sample-webhook/main.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		_, err := w.Write(topic)
+		if err != nil {
+			fmt.Fprintf(w, `{"error":"failed to pull topics, cause: %s"}`, err)
+		}
+	case <-time.After(topicTimeout):
+		fmt.Fprintf(w, `{"error":"no topic found in queue"}`)
+	}
+}
+
+func main() {
```
Comment (llorllale) [inline cmd/sample-webhook/main.go:52]: doesn't seem related to openapi

File: Makefile (hunk 1 above)
Comment (soluchok) [reply Makefile:91]: I assume it is part of the dock.

**Review by llorllale — APPROVED** [review]
This PR adds openapi spec generation as well as the OpenAPI demo.

File: cmd/sample-webhook/main.go (hunk 2 above)
Comment (soluchok) [reply cmd/sample-webhook/main.go:52]: I will remove it.

File: Makefile (hunk 1 above)
Comment (llorllale) [reply Makefile:91]: @soluchok it is likely needed for the demo, but you can confirm

File: pkg/controller/rest/didclient/models.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (Baha-sk) [inline pkg/controller/rest/didclient/models.go:1]: will these models ever be used in this repo?

**Review by Baha-sk — APPROVED** [review]

File: pkg/controller/rest/didclient/models.go (hunk 3 above)
Comment (soluchok) [reply pkg/controller/rest/didclient/models.go:1]: Yes, they are part of the spec.

File: cmd/sample-webhook/main.go (hunk 2 above)
Comment (soluchok) [reply cmd/sample-webhook/main.go]: Done!

**Review by sudeshrshetty — APPROVED** [review]
